src/services/identity/LaunchClient.ts

```diff
@@ -142,13 +168,33 @@ class LaunchClient {
     return subDomains
   }
 
+  async sendDeleteDomainAssociationCommand(
+    input: DeleteDomainAssociationCommandInput
+  ): Promise<void> {
+    if (this.shouldMockAmplifyDomainCalls()) {
+      this.mockDeleteDomainAssociationOutput(input)
+    }
+    await this.amplifyClient.send(new DeleteDomainAssociationCommand(input))
+  }
+
+  mockDeleteDomainAssociationOutput(
+    input: DeleteDomainAssociationCommandInput
+  ) {
+    if (!this.mockDomainAssociations.has(input.domainName)) {
+      throw new AmplifyError(
+        `NotFoundException: Domain association ${input.domainName} not found.`
```

Comment:
This mocks the behaviour of the real amplify client and throws an exception if domain hasnt been created yet